x-pack/plugins/security_solution/public/detections/components/alerts_table/timeline_actions/use_investigate_in_timeline.tsx

```diff
@@ -35,6 +38,49 @@ interface UseInvestigateInTimelineActionProps {
   onInvestigateInTimelineAlertClick?: () => void;
 }
 
+const detectionExceptionList = (ecsData: Ecs): ExceptionListId[] => {
+  let exceptionsList = getField(ecsData, ALERT_RULE_EXCEPTIONS_LIST) ?? [];
+  let detectionExceptionsList: ExceptionListId[] = [];
+  try {
+    if (Array.isArray(exceptionsList) && exceptionsList.length === 0) {
+      const ruleParameters = getField(ecsData, ALERT_RULE_PARAMETERS) ?? {};
+      if (ruleParameters.length > 0) {
+        const parametersObject = JSON.parse(ruleParameters[0]);
+        exceptionsList = parametersObject?.exceptions_list ?? [];
+      }
+    } else if (exceptionsList && exceptionsList.list_id) {
+      return exceptionsList.list_id
+        .map((listId: string, index: number) => {
+          const type = exceptionsList.type[index];
+          return {
+            exception_list_id: listId,
+            namespace_type: exceptionsList.namespace_type[index],
+            type,
+          };
+        })
+        .filter((exception: ExceptionListIdentifiers) => exception.type === 'detection');
```

Comment:
Not necessarily necessary, but could be good to use the enum here? A lot of places don't use it already though 🤷🏾‍♂️  https://github.com/elastic/kibana/blob/ee0b1ec78f9aeec081e77882527455ef734bc60e/packages/kbn-securitysolution-io-ts-list-types/src/common/exception_list/index.ts#L24